lms/extractors/ziparchive.py

```diff
@@ -30,35 +30,60 @@ def can_extract(self) -> bool:
         return self.is_zipfile
 
     @staticmethod
-    def _extract(archive: ZipFile, filename: str) -> File:
+    def _extract(archive: ZipFile, filename: str, dirname: str = '') -> File:
         with archive.open(filename) as current_file:
             log.debug(f'Extracting from archive: {filename}')
             code = current_file.read()
         decoded = code.decode('utf-8', errors='replace').replace('\x00', '')
-        return File(path=f'/{filename}', code=decoded)
-
-    def get_exercise(self, file: ZipFile) -> Tuple[int, List[File]]:
+        return File(path=f'/{filename.lstrip(dirname)}', code=decoded)
```

Comment:
Please read about what `str.lstrip` does.

When we'll move to Python 3.9 it will be possible to use `str.removeprefix`, until then we remain with good ol':

```python
if filename.startswith(dirname):
    filename = filename[:len(dirname)]
```

Reply:
BTW, why not removing the dirname in `get_files`?

It seems to me that it's not `_extract`'s responsibility to take

Reply:
Probably you meant - filename[len(dirname):]

Reply:
Can't really do it in `get_files` cause I open it with archive in `_extract`